src/daemon/collector/platforms/facebook/handleApiResponse.js

```diff
@@ -38,8 +38,9 @@ const containsSponsoredResponse = (response) => {
   return (
     response.category === "SPONSORED" ||
     // 2024-08-25: Category has since been encoded as field `category_enc`
-    //  Try to detect based on presence of sponsored_data content
+    // Try to detect based on presence of sponsored_data content
     _.has(response, "node.sponsored_data.ad_id") ||
+    _.has(response, "node.sponsored_data") || // 2024-10-01 changes: there is only node.sponsored_data, not necessarily ad_id
```

Comment:
really? what is in ther if ther's no ad_id? if it's empty, then it isn't an ad

Reply:
The issue is that `ad_id` is not longer at the top level of `sponsored_data` and there are also multiple nested keys called `sponsored_data`. In this case, we are checking just for `sponsored_data` as a key and then we look for `ad_id`